pkg/receive/expandedpostingscache/cache.go

```diff
@@ -249,13 +249,17 @@ func isHeadBlock(blockID ulid.ULID) bool {
 }
 
 func metricNameFromMatcher(ms []*labels.Matcher) (string, bool) {
+	var metricName string
 	for _, m := range ms {
 		if m.Name == labels.MetricName && m.Type == labels.MatchEqual {
-			return m.Value, true
+			if metricName != "" {
+				return "", false
```

Comment:
The intention here is to return `"", false` if two `Matcher`s match the given metric name?

Reply:
The purpose is to not cache if there more than two matchers on `__name__`. Technically such queries don't make sense and will never return results but just in case to avoid filling the cache with empty items